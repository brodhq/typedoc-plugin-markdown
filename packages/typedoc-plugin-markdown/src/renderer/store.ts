import { ProjectReflection } from 'typedoc';

import { PluginOptions, State } from './types';

const state: State = {
  project: new ProjectReflection(''),
  currentModel: null,
  hasReadme: null,
  currentUrl: '',
  projectName: '',
  globalsName: '',
  projectUrl: '',
  options: {},
};

export const useState = () => state;

export function setDefaultState(
  project: ProjectReflection,
  options: Partial<PluginOptions>,
) {
  setState({
    projectName: project.name,
    projectUrl: project.url,
    project: project,
    globalsName:
      options.entryPoints && options.entryPoints.length > 1
        ? 'Modules'
        : 'Exports',
    options: options,
    hasReadme: options.readme !== 'none',
  });
}

export function setState(nextState) {
  Object.assign(state, nextState);
}
