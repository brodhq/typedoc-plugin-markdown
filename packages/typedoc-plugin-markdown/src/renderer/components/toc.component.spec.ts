import { TestApp } from '../../../test/test-app';
import { setState } from '../store';
import { TocComponent } from './toc.component';

describe(`Toc Component`, () => {
  let testApp: TestApp;

  beforeAll(async () => {
    testApp = new TestApp(['categories.ts', 'toc.ts']);
    await testApp.bootstrap();
  });

  describe(`(with hideInPageTOC=false)`, () => {
    beforeAll(async () => {
      setState({ options: { hideInPageTOC: false } });
    });

    test(`should compile module toc'`, () => {
      const model = testApp.getPageByModelName('toc').model;
      expect(TocComponent(model)).toMatchSnapshot();
    });

    test(`should compile module toc with categories'`, () => {
      const model = testApp.getPageByModelName('categories').model;
      expect(TocComponent(model)).toMatchSnapshot();
    });

    test(`should compile interface toc'`, () => {
      const model = testApp.getPageByModelName('Interface1').model;
      expect(TocComponent(model)).toMatchSnapshot();
    });
  });

  describe(`(with hideInPageTOC=true)`, () => {
    beforeAll(async () => {
      setState({ options: { hideInPageTOC: true } });
    });

    test(`should compile basic groups'`, () => {
      const model = testApp.getPageByModelName('toc').model;
      expect(TocComponent(model)).toMatchSnapshot();
    });

    test(`should compile category groups'`, () => {
      const model = testApp.getPageByModelName('categories').model;
      expect(TocComponent(model)).toMatchSnapshot();
    });

    test(`should compile interface toc'`, () => {
      const model = testApp.getPageByModelName('Interface1').model;
      expect(TocComponent(model)).toBe('');
    });
  });
});
