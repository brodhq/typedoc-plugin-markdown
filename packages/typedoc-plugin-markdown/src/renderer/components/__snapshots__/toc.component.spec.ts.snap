// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toc Component (with hideInPageTOC=false) should compile interface toc' 1`] = `
"## Table of contents

### Properties

- [x](toc.interface1.md#x)"
`;

exports[`Toc Component (with hideInPageTOC=false) should compile module toc with categories' 1`] = `
"## Table of contents

### CategoryA Interfaces

- [CategoryInterfaceA1](../interfaces/categories.categoryinterfacea1.md)
- [CategoryInterfaceA2](../interfaces/categories.categoryinterfacea2.md)

### Other Interfaces

- [OtherInterfaceA](../interfaces/categories.otherinterfacea.md)

### CategoryA Variables

- [categoryVariableA1](categories.md#categoryvariablea1)
- [categoryVariableA2](categories.md#categoryvariablea2)

### CategoryB Variables

- [categoryVariableB1](categories.md#categoryvariableb1)

### Other Variables

- [otherVariableA](categories.md#othervariablea)
- [otherVariableB](categories.md#othervariableb)

### CategoryA Functions

- [categoryFunctionA1](categories.md#categoryfunctiona1)
- [categoryFunctionA2](categories.md#categoryfunctiona2)
- [categoryFunctionA3](categories.md#categoryfunctiona3)

### CategoryB Functions

- [categoryFunctionB2](categories.md#categoryfunctionb2)

### Other Functions

- [otherFunctionA](categories.md#otherfunctiona)
- [otherFunctionB](categories.md#otherfunctionb)"
`;

exports[`Toc Component (with hideInPageTOC=false) should compile module toc' 1`] = `
"## Table of contents

### Classes

- [Class1](../classes/toc.class1.md)
- [Class2](../classes/toc.class2.md)

### Interfaces

- [Interface1](../interfaces/toc.interface1.md)
- [Interface2](../interfaces/toc.interface2.md)

### Variables

- [variable1](toc.md#variable1)
- [variable2](toc.md#variable2)

### Functions

- [function1](toc.md#function1)
- [function2](toc.md#function2)"
`;

exports[`Toc Component (with hideInPageTOC=true) should compile basic groups' 1`] = `
"## Classes

- [Class1](../classes/toc.class1.md)
- [Class2](../classes/toc.class2.md)

## Interfaces

- [Interface1](../interfaces/toc.interface1.md)
- [Interface2](../interfaces/toc.interface2.md)"
`;

exports[`Toc Component (with hideInPageTOC=true) should compile category groups' 1`] = `
"## CategoryA Interfaces

- [CategoryInterfaceA1](../interfaces/categories.categoryinterfacea1.md)
- [CategoryInterfaceA2](../interfaces/categories.categoryinterfacea2.md)

## Other Interfaces

- [OtherInterfaceA](../interfaces/categories.otherinterfacea.md)"
`;
